Model/Builders/ServiceOrderBuilder.php

```diff
@@ -205,66 +216,54 @@ protected function rstrpos($haystack, $needle, $offset = null)
+                verified_terms_of_service: $verifiedTermsOfService
+            )
+        );
     }
 
     /**
      * @param $order
      * @param $paymentDetails
      * @param $customerData
-     *
+     * @param $urlProvider
      * @return Order
      */
-    public function collectData( $order, $paymentDetails, $customerData)
-    {
-        return new Order(
-            currency:  new Currency('EUR'),// new Currency($payment_amount->getCurrencyCode()),
-            amount: new Amount(floatval(1)*100),
-            transactions: $paymentDetails,
-            customer: $customerData
-        );
-    }
-
-    /**
-     * Collect data for order
-     *
-     * @return array
-     */
-
-    public function collectDataForOrder($order, $methodCode, $urlProvider, $orderLines, $paymentDetails, $customerData = null)
+    public function collectData($order, $paymentDetails, $customerData, $urlProvider, $paymentMethodDetails)
     {
-        $orderData = array_filter([
-            'amount' => $this->configRepository->getAmountInCents((float)$order->getBaseGrandTotal()),
-            'currency' => $order->getOrderCurrencyCode(),
-            'description' => $this->configRepository->getDescription($order, $methodCode),
-            'merchant_order_id' => $order->getIncrementId(),
-            'return_url' => $urlProvider->getReturnUrl(),
-            'webhook_url' => $urlProvider->getWebhookUrl(),
-            'transactions' => $paymentDetails,
-            'extra' => $this->getExtraLines(),
-            'order_lines' => $orderLines->get($order),
-            'customer' => $customerData
-        ]);
-        return $orderData;
+            return new Order(
+                currency: new Currency($order->getOrderCurrencyCode()),
+                amount: new Amount( (int)($order->getBaseGrandTotal())),
+                transactions: new Transactions(
+                    new Transaction(
+                        paymentMethodDetails: $paymentMethodDetails,
+                        paymentMethod: $paymentDetails
+                    )
+                ),
+                customer: new Customer(
+                    additionalAddresses: new AdditionalAddresses(
+                        new Address(
+                            addressType: $customerData['address_type'],
+                            postalCode: $customerData['postal_code'],
+                            country: new Country($customerData['country'])
+                        ),
+                    ),
+                    firstName: $customerData['first_name'],
+                    lastName: $customerData['last_name'],
+                    emailAddress: new EmailAddress($customerData['email_address']),
+                    phoneNumbers: new PhoneNumbers(),
+                    country: new Country($customerData['country']),
+                    locale: new Locale($customerData['locale']),
+                    merchantCustomerId: $customerData['merchant_customer_id']
+                ),
```

Comment:
Is it possible to create method `getCustomer` and move this code there. It will make code more readable and maintainable